Metalama.Framework.Engine/Linking/AspectLinkerDiagnosticDescriptors.cs

```diff
@@ -20,6 +22,30 @@ public static class AspectLinkerDiagnosticDescriptors
                 _category,
                 Error );
 
+        internal static readonly DiagnosticDefinition<(StatementSyntax Statement, INamedType Type)>
+            CantAddStatementToPrimaryConstructor = new(
+                "LAMA0651",
+                "Statement can't be added as an initializer to primary constructor.",
+                "The statement '{0}' can't be added as an initializer to a primary constructor on type '{1}'. Only simple assignment is supported.",
+                _category,
+                Error );
+
+        internal static readonly DiagnosticDefinition<(string Expression, INamedType Type, string Explanation)>
+            CantAssignToExpressionFromPrimaryConstructor = new(
+                "LAMA0652",
+                "Expression can't be used as the assignment target for an initializer of a primary constructor.",
+                "The expression '{0}' can't be used as the assignment target for an initializer of a primary constructor on type '{1}' {2}.",
+                _category,
+                Error );
+
+        internal static readonly DiagnosticDefinition<(DeclarationKind DeclarationKind, IFieldOrProperty Member, INamedType Type)>
+            CantAssignToMemberMoreThanOnceFromPrimaryConstructor = new(
+                "LAMA0653",
+                "Member can't be used as the assignment target for an initializer of a primary constructor more than once.",
+                "The {0} '{1}' can't be used as the assignment target for an initializer of a primary constructor on type '{2}' more than once.",
```

Comment:
Is it an error though? I think it can be safe to just replace the existing expression (maybe with some "force" option). Regardless, it would be better to include aspect name in the diagnostic, otherwise it's not very helpful to the user.

Reply:
I don't like silently dropping existing code, especially code that can potentially have side-effects. Having a "force" option is reasonable, but I think that's outside the scope now.

I have improved the error message.